lib/season_stats.rb

```diff
@@ -5,7 +5,49 @@ def initialize(locations)
     super(locations)
   end
 
-  # add winningest and worst coach methods
+  def winningest_coach(season)
+    coach_games = Hash.new([0,0,0])
+    game_teams_by_season(season).each do |game_team|
+      coach_games[game_team.head_coach] = [0,0,0]
+    end
+    game_teams_by_season(season).each do |game_team|
+      if game_team.result == "WIN"
+        coach_games[game_team.head_coach][0] +=1 
+      elsif game_team.result == "TIE"
+        coach_games[game_team.head_coach][1] +=1
+      elsif game_team.result == "LOSS"
+      coach_games[game_team.head_coach][2] +=1
+      else 
+        return false
+      end
+    end
```

Comment:
Nice use of enumerables to go through the data and find the best and worst coaches.